test: failure in `mining_basic.py` AssertionError: not(4.656542373906924E-10 == 4.656542373906925E-10)

x86_64 Alpine Linux
master @ c779ee3a4044df3a263394bbb8b104aeeaa7c727
```bash
/bitcoin # build/test/functional/test_runner.py mining_basic.py --valgrind --timeout-factor=0
Temporary test directory at /tmp/test_runner_₿_🏃_20250515_142647
Remaining jobs: [mining_basic.py]
1/1 - mining_basic.py failed, Duration: 58 s                                                                        

stdout:
2025-05-15T14:26:48.007000Z TestFramework (INFO): PRNG seed is: 1679234749093477821
2025-05-15T14:26:48.008000Z TestFramework (INFO): Initializing test directory /tmp/test_runner_₿_🏃_20250515_142647/mining_basic_0
2025-05-15T14:26:57.860000Z TestFramework (INFO): Create some old blocks
2025-05-15T14:27:19.347000Z TestFramework (INFO): test blockversion
2025-05-15T14:27:45.954000Z TestFramework (INFO): getmininginfo
2025-05-15T14:27:46.014000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/bitcoin/test/functional/test_framework/test_framework.py", line 183, in main
    self.run_test()
  File "/bitcoin/build/test/functional/mining_basic.py", line 392, in run_test
    assert_equal(mining_info['difficulty'], Decimal('4.656542373906925E-10'))
  File "/bitcoin/test/functional/test_framework/util.py", line 77, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(4.656542373906924E-10 == 4.656542373906925E-10)
2025-05-15T14:27:46.067000Z TestFramework (INFO): Not stopping nodes as test failed. The dangling processes will be cleaned up later.
2025-05-15T14:27:46.067000Z TestFramework (WARNING): Not cleaning up dir /tmp/test_runner_₿_🏃_20250515_142647/mining_basic_0
2025-05-15T14:27:46.067000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/test_runner_₿_🏃_20250515_142647/mining_basic_0/test_framework.log
2025-05-15T14:27:46.067000Z TestFramework (ERROR): 
2025-05-15T14:27:46.067000Z TestFramework (ERROR): Hint: Call /bitcoin/test/functional/combine_logs.py '/tmp/test_runner_₿_🏃_20250515_142647/mining_basic_0' to consolidate all logs
2025-05-15T14:27:46.067000Z TestFramework (ERROR): 
2025-05-15T14:27:46.067000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2025-05-15T14:27:46.068000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2025-05-15T14:27:46.068000Z TestFramework (ERROR):
```

Verbose logs are available here: https://gist.github.com/fanquake/e2f2e4eba103d883e23801b154af45f2

--- a/test/functional/mining_basic.py
+++ b/test/functional/mining_basic.py
@@ -390,14 +390,13 @@ class MiningTest(BitcoinTestFramework):
         assert 'currentblockweight' not in mining_info
         assert_equal(mining_info['bits'], nbits_str(REGTEST_N_BITS))
         assert_equal(mining_info['target'], target_str(REGTEST_TARGET))
-        assert_equal(mining_info['difficulty'], Decimal('4.656542373906925E-10'))
-        assert_equal(mining_info['next'], {
-            'height': 201,
-            'target': target_str(REGTEST_TARGET),
-            'bits': nbits_str(REGTEST_N_BITS),
-            'difficulty': Decimal('4.656542373906925E-10')
-        })
-        assert_equal(mining_info['networkhashps'], Decimal('0.003333333333333334'))
+        # We don't care about precision, round to avoid mismatch under Valgrind:
+        assert_equal(round(mining_info['difficulty'], 10), Decimal('0.0000000005'))
+        assert_equal(mining_info['next']['height'], 201)
+        assert_equal(mining_info['next']['target'], target_str(REGTEST_TARGET))
+        assert_equal(mining_info['next']['bits'], nbits_str(REGTEST_N_BITS))
+        assert_equal(round(mining_info['next']['difficulty'], 10), Decimal('0.0000000005'))
+        assert_equal(round(mining_info['networkhashps'], 5), Decimal('0.00333'))
         assert_equal(mining_info['pooledtx'], 0)
 
         self.log.info("getblocktemplate: Test default witness commitment")

